### PR TITLE
Add RISC-V PMP Driver for Memory Protection

### PR DESCRIPTION
**Pull request by SurajSonawane2415**
- Implement PMP initialization and configuration functions
- Support for TOR, NA4, and NAPOT entry types
- Provide utility functions for PMP register access

This driver enables memory protection on RISC-V systems to enhance security. Tested with the riscv/qemu template.

Please review and suggest any required changes.

### CODE REVIEW

File: src/arch/riscv/pmp/pmp.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+static inline void write_pmpaddr(uint32_t reg, uint32_t value) {
+    REG32_STORE(PMP_ADDR_BASE + reg * 4, value);
+}
+
+/**
+ * Initializes the PMP subsystem by clearing all PMP entries and setting up 
+ * predefined PMP regions based on configuration options.
+ *
+ */
+int pmp_init(void) {
```
Comment (anton-bondarev) [inline src/arch/riscv/pmp/pmp.c:39]: Does it call with UNIT_INIT or directly from code? If it call from code where it will be?

Comment (SurajSonawane2415) [reply src/arch/riscv/pmp/pmp.c:39]: Sorry, I forgot to call it, but I will do it now using UNIT_INIT.

**Review by anton-bondarev — APPROVED** [review]
LGTM

**Comment by SurajSonawane2415** [comment]
Thank you @anton-bondarev.

**Pull request merged**